File: 121. Best Time to Buy and Sell Stock.go
package main

func max(x, y int) int {
	if x > y {
		return x
	}
	return y
}

func maxProfit(prices []int) int {

	i := 0
	j := 0
	profit := 0

	for j < len(prices) {
		profit = max(prices[j]-prices[i], profit)

		if prices[i] > prices[j] {
			i++
		} else {
			j++
		}
	}

	return profit
}

File: 121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        result = -sys.maxsize
        p1 = p2 = 0

        while p2 < len(prices):
            
            result = max(result,prices[p2]-prices[p1])
            if prices[p1] > prices[p2]: p1 += 1
            else: p2 += 1

        return result
